src/filtered-list/index.ts

```diff
@@ -23,6 +25,17 @@ class FilteredList {
     this.updateItems();
   }
 
+  onKeyDown(event, value: string = this.refs.filter.value) {
+    value = value.trim().toLowerCase();
+    if (event.keyCode === RETURN_KEY_CODE) {
+      const foundItem = this.props.items.find((item) => typeof item === 'object' && item.value && item.value.toLowerCase() === value);
+       
+      if (typeof foundItem === 'object' && typeof foundItem.onClick === 'function') {
```

Comment:
should be `if (foundItem && typeof (<ItemObject>foundItem).onClick === 'function')`